src/__snapshots__/middleware.test.js.snap

```diff
@@ -0,0 +1,2011 @@
+      Object {
+        "meta": "successMeta",
+        "payload": Object {
+          "username": "Alice",
+        },
+        "type": "SUCCESS",
+      },
+    ],
+  ],
+  "results": Array [
+    Object {
+      "isThrow": false,
+      "value": Object {
+        "meta": "requestMeta",
+        "payload": "requestPayload",
+        "type": "REQUEST",
+      },
+    },
+    Object {
+      "isThrow": false,
+      "value": Object {
+        "meta": "successMeta",
+        "payload": Object {
+          "username": "Alice",
+        },
+        "type": "SUCCESS",
+      },
+    },
+  ],
+}
+`;
+
+exports[`#apiMiddleware must dispatch a success FSA on a successful API call with an empty JSON response: fetch mock 1`] = `
+Object {
+  "calls": Array [
+    Array [
+      "http://127.0.0.1/api/users/1",
+      Object {
+        "body": undefined,
+        "credentials": undefined,
+        "headers": Object {},
+        "method": "GET",
+      },
+    ],
+  ],
+  "instances": Array [
+    undefined,
+  ],
+  "invocationCallOrder": Any<Object>,
+  "results": Array [
+    Object {
+      "isThrow": false,
+      "value": Promise {},
+    },
+  ],
+}
+`;
+
+exports[`#apiMiddleware must dispatch a success FSA on a successful API call with an empty JSON response: final result 1`] = `
+Object {
+  "meta": "successMeta",
+  "payload": Object {},
+  "type": "SUCCESS",
+}
+`;
+
+exports[`#apiMiddleware must dispatch a success FSA on a successful API call with an empty JSON response: next mock 1`] = `
+[MockFunction] {
+  "calls": Array [
+    Array [
+      Object {
+        "meta": "requestMeta",
+        "payload": "requestPayload",
+        "type": "REQUEST",
+      },
+    ],
+    Array [
+      Object {
+        "meta": "successMeta",
+        "payload": Object {},
+        "type": "SUCCESS",
+      },
+    ],
+  ],
+  "results": Array [
+    Object {
+      "isThrow": false,
+      "value": Object {
+        "meta": "requestMeta",
+        "payload": "requestPayload",
+        "type": "REQUEST",
+      },
+    },
+    Object {
+      "isThrow": false,
+      "value": Object {
+        "meta": "successMeta",
+        "payload": Object {},
+        "type": "SUCCESS",
+      },
+    },
+  ],
+}
+`;
+
+exports[`#apiMiddleware must dispatch a success FSA with an error state on a successful API call with an invalid JSON response: fetch mock 1`] = `
+Object {
+  "calls": Array [
+    Array [
+      "http://127.0.0.1/api/users/1",
+      Object {
+        "body": undefined,
+        "credentials": undefined,
+        "headers": Object {},
+        "method": "GET",
+      },
+    ],
+  ],
+  "instances": Array [
+    undefined,
+  ],
+  "invocationCallOrder": Any<Object>,
+  "results": Array [
+    Object {
+      "isThrow": false,
+      "value": Promise {},
+    },
+  ],
+}
+`;
+
+exports[`#apiMiddleware must dispatch a success FSA with an error state on a successful API call with an invalid JSON response: final result 1`] = `
+Object {
+  "error": true,
```

Comment:
Unrelated -- but I want to note that there are still a few situations in which REQUEST and SUCCESS type actions dispatch with an error flag

I think a follow up to #175 is in order